deepmd/utils/tabulate.py

```diff
@@ -0,0 +1,252 @@
+import re
+import math
+import numpy as np
+from tqdm import tqdm
+from typing import Tuple, List
+from deepmd.env import tf
+from deepmd.env import op_module
+from tensorflow.python.platform import gfile
+from tensorflow.python.framework import tensor_util
+
+class DeepTabulate():
+    """
+    Class for tabulation.
+    Compress a model, which including tabulating the embedding-net. 
+    The table is composed of fifth-order polynomial coefficients and is assembled from two sub-tables. The first table takes the stride(parameter) as it\'s uniform stride, while the second table takes 10 * stride as it\s uniform stride 
+    The range of the first table is automatically detected by deepmd-kit, while the second table ranges from the first table\'s upper boundary(upper) to the extrapolate(parameter) * upper.
+    """
+    def __init__(self,
+                 model_file,
+                 data_type,
```

Comment:
`precision` would be consistent with other part of the code when referring to the precision of floating point numbers.

Reply:
The precision would be consistent with the se_a. If we do model compression and precision conversion at the same time, we need this parameter.

Reply:
Leave the precision parameter in the descriptor(se_a), the precision of class DeepTabulate is consistent with the input model precision